Interactive Partials Examples Fixes #18

papasmile opened the pull request:
* All examples Interactive
* Tidied up into examples/partials/ folder
* Since no content changed, also updated Chinese; don't speak Mandarin yet so would be nice if someone that does could spot check...

Should fully address issue #18.  Tested locally with `npm run preview`.

Just in time for Hack...-ruary...?  Anyway there you go :)

nknapp commented:
Wow, thanks. Though, I would still like to have some changes :wink: ...

1. You can use the "partials:"-property in the YAML-frontmatter to define partials. Not need to add a preparation-script. Have a look at https://handlebarsjs.com/examples/all-features.html on how to define them and at https://handlebarsjs.com/contributing/interactive-examples.html#embedding-parts-of-an-example on how to include the partials in the text.
**Exceptions are "examples/partails/register.md" and "examples/partials/basic.md". They should use a preparation-script because their purpose is to show how to register partials.**
2. Maybe you can reuse "register.md" where you created "basic.md".
3. Optional: My original thought was that all examples should use variations of a common data-set (i.e. [this one](https://github.com/handlebars-lang/docs/blob/master/src/examples/_example-base-data.yaml)). It would be cool to use that data. If you think additional (but related) data would help with the example, just add it to that file. "It would be cool" means that it is not strictly necessary though. It can be done later. 

Please apply 1) and see if 2) is a good idea. I will merge and then have a look at the preview site. It is easier to review there.

papasmile commented:
😎 
1) Nice, had missed that feature for partials... updated!
2) Looked at register.md but it is using parameters; that is covered in a separate section so thinking keeping basic.md would be clearer.
3) Agreed, I went ahead and merged with _example_base-data.yml in examples where input was less relevant to actual code.

Also updated ZH docs with same.

papasmile commented:
Build passed but something was up with Github a bit ago so stuck showing yellow....

nknapp commented:
I added some comments (I hope it worked from my mobile) in places I found a "registerPartials" that should be replaced.

papasmile commented:
Ah, I had updated partials.md but not those extra example entires.  Very well, then, updated!

nknapp commented on src/examples/partials/variable.md:
```diff
@@ -0,0 +1,10 @@
+example:
+  template: |
+    {{> (lookup . 'myVariable') }}
+  preparationScript: |
```
Sorry, still found one. In the chinese version too...

nknapp commented:
@papasmile I merged your changes. Would be nice anyway if you took care of the `variable.md` example. I invited you to a team that gives you access to the master branch of this repository. 

In case you want to do more changes to the docs. Changes on master will appear on the preview site at https://handlebars-preview.knappi.org. I still have to trigger the deploy to handlebarsjs.com myself.

Thanks for all the work that you have done so far.

papasmile commented:
Wow that is very cool of you, thanks!

I went ahead and updated variables.md and verified on preview:
https://github.com/handlebars-lang/docs/commit/4fb007398fe3c899dd26ae9fa3ba4499e7fe179b
https://github.com/handlebars-lang/docs/commit/6f5b71d836a064c598d81144a032c4d504db1e50

nknapp commented:
Actually, it is just saving me some work if you can push to master directly. And maybe you can even review PRs from other people, if you like. 

I still can't promise to put changes to prod quickly, because I cannot to this on my mobile phone.

papasmile commented:
Yes, those commits are on master and ready to go.  Take your time!
